src/CloudBees.OpenFeature.Provider/CloudBeesProvider.cs

```diff
@@ -5,24 +5,36 @@
 using Io.Rollout.Rox.Core.Context;
 using Io.Rollout.Rox.Server;
 using OpenFeature;
+using OpenFeature.Constant;
 using OpenFeature.Model;
 
 namespace CloudBees.OpenFeature.Provider
 {
     public class CloudBeesProvider : FeatureProvider
     {
         private readonly Metadata _metadata = new Metadata("CloudBees Provider");
+        private ProviderStatus _status = ProviderStatus.NotReady;
+        private string _appKey;
 
-        /// <summary>
-        /// Call this once before using the provider.
-        /// </summary>
-        /// <param name="appKey">The application key from app.rollout.io</param>
-        public static async Task Setup(string appKey) => await Rox.Setup(appKey);
+        public CloudBeesProvider(string appKey)
+        {
+            _appKey = appKey ?? throw new ArgumentNullException(nameof(appKey));
+        }
+
+        public override async Task Initialize(EvaluationContext context)
+        {
+            var options = new RoxOptions(new RoxOptions.RoxOptionsBuilder());

```

Comment:
there's no need to initialize RoxOptions here, can use the Setup(key) overload
anyway... it doesn't matter

i think we should allow sending roxOptions to our provider in the future
not sure if the openfeature standard allow it, but we already do in node for ex (https://github.com/rollout/cloudbees-openfeature-provider-node/blob/main/src/provider.ts#L10)